File: OpsNext/packages/shared/src/types/tenant.ts
export type SubscriptionTier = 'BASIC' | 'PROFESSIONAL' | 'ENTERPRISE';
export type TenantStatus = 'ACTIVE' | 'SUSPENDED' | 'DELETED';

export interface Tenant {
  id: string;
  slug: string;
  name: string;
  status: TenantStatus;
  subscriptionTier: SubscriptionTier;
  schemaName: string;
  createdAt: Date;
  updatedAt: Date;
  deletedAt?: Date | null;
}

export interface TenantConfig {
  id: string;
  tenantId: string;
  logoUrl?: string | null;
  primaryColor?: string | null;
  timezone: string;
  currency: string;
  language: string;
  dateFormat: string;
  featureFlags: Record<string, boolean>;
}

export interface CreateTenantInput {
  name: string;
  slug?: string;
  industry?: string;
  size?: string;
  timezone?: string;
  currency?: string;
  language?: string;
  subscriptionTier?: SubscriptionTier;
}

File: OpsNext/backend/src/main/java/io/opsnext/api/platform/entity/Tenant.java
package io.opsnext.api.platform.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.io.Serializable;
import java.time.Instant;

@Entity
@Table(name = "tenants", schema = "public")
@Getter
@Setter
public class Tenant implements Serializable {

    @Id
    @Column(length = 36)
    private String id;

    @Column(unique = true, nullable = false, length = 63)
    private String slug;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private TenantStatus status = TenantStatus.ACTIVE;

    @Column(nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private SubscriptionTier subscriptionTier = SubscriptionTier.BASIC;

    @Column(nullable = false)
    private String schemaName;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;

    private Instant deletedAt;

    @OneToOne(mappedBy = "tenant", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private TenantConfig config;

    public enum TenantStatus { ACTIVE, SUSPENDED, DELETED }
    public enum SubscriptionTier { BASIC, PROFESSIONAL, ENTERPRISE }
}
